src/imgui/ImGuiMachine.cc

```diff
@@ -31,6 +31,36 @@ using namespace std::literals;
 
 namespace openmsx {
 
+void ImGuiMachine::save(ImGuiTextBuffer& buf)
+{
+	savePersistent(buf, *this, persistentElements);
+
+	for (const auto& item : recentMachines) {
+		if (!item.empty()) {
+			buf.appendf("machine.name=%s\n", item.c_str());
+		}
+	}
+}
+
+static void addRecentItem(circular_buffer<std::string>& recentItems, const std::string& item)
+{
+	if (auto it = ranges::find(recentItems, item); it != recentItems.end()) {
+		// was already present, move to front
+		std::rotate(recentItems.begin(), it, it + 1);
+	} else {
+		// new entry, add it, but possibly remove oldest entry
+		if (recentItems.full()) recentItems.pop_back();
+		recentItems.push_front(item);
+	}
+}
```

Comment:
This function seems like a good candidate to share. Maybe generalize it to
```
template<typename T>
void addRecentItem(circular_buffer<T>& buffer, const T& item)
{
...
}
```
And then call it from here and from ImGuiMedia.

Maybe do this is a separate commit?

Reply:
Done. I made a commit before the actual commit to isolate this and then use it in the actual commit.